src/reference/asciidoc/metrics.adoc

```diff
@@ -0,0 +1,269 @@
+----
+<int:management
+    default-logging-enabled="false" <1>
+    default-counts-enabled="false" <2>
+    default-stats-enabled="false" <3>
+    counts-enabled-patterns="foo, !baz, ba*" <4>
+    stats-enabled-patterns="fiz, buz" <5>
+    metrics-factory="myMetricsFactory" /> <6>
+----
+
+[source, java]
+----
+@Configuration
+@EnableIntegration
+@EnableIntegrationManagement(
+		defaultLoggingEnabled = "false", <1>
+		defaultCountsEnabled = "false", <2>
+		defaultStatsEnabled = "false", <3>
+		countsEnabled = { "foo", "${count.patterns}" }, <4>
+		statsEnabled = { "qux", "!*" }, <5>
+		MetricsFactory = "myMetricsFactory") <6>
+public static class ContextConfiguration {
+...
+}
+----
+
+<1> Set to `false` to disable all logging in the main message flow, regardless of the log system category settings.
+Set to 'true' to enable debug logging (if also enabled by the logging subsystem).
+
+<2> Enable or disable count metrics for components not matching one of the patterns in <4>.
+
+<3> Enable or disable statistical metrics for components not matching one of the patterns in <5>.
+
+<4> A comma-delimited list of patterns for beans for which counts should be enabled; negate the pattern with `!`.
+First match wins (positive or negative).
+In the unlikely event that you have a bean name starting with `!`, escape the `!` in the pattern: `\!foo` positively
+matches a bean named `!foo`.
+
+<5> A comma-delimited list of patterns for beans for which statistical metrics should be enabled; negate the pattern
+with `!`.
+First match wins (positive or negative).
+In the unlikely event that you have a bean name starting with `!`, escape the `!` in the pattern: `\!foo` positively
+matches a bean named `!foo`.
+Stats implies counts.
+
+<6> A reference to a `MetricsFactory`.
+See <<mgmt-metrics-factory>>.
+
+At runtime, counts and statistics can be obtained by calling `IntegrationManagementConfigurer` `getChannelMetrics`,
+`getHandlerMetrics` and `getSourceMetrics`, returning `MessageChannelMetrics`, `MessageHandlerMetrics` and
+`MessageSourceMetrics` respectively.
+
+See the javadocs for complete information about these classes.
+
+When JMX is enabled (see <<jmx>>), these metrics are also exposed by the `IntegrationMBeanExporter`.
+
+[[mgmt-channel-features]]
+==== MessageChannel Metric Features
+
+Message channels report metrics according to their concrete type.
+If you are looking at a `DirectChannel`, you will see statistics for the send operation.
+If it is a `QueueChannel`, you will also see statistics for the receive operation, as well as the count of messages that are currently buffered by this `QueueChannel`.
+In both cases there are some metrics that are simple counters (message count and error count), and some that are estimates of averages of interesting quantities.
+The algorithms used to calculate these estimates are described briefly in the section below.
+
+.MessageChannel Metrics
+
+
+[cols="1,2,3", options="header"]
+|===
+| Metric Type
```

Comment:
I like the pdf view for this table, but wonder if we can improve it for HTML. This caption isn't so long, but it is wrapped making the head of table higher.
So, can we some how adjust the column width?